File: cs25backend/src/main/java/edu/dongguk/cs25backend/security/CustomUserDetail.java
package edu.dongguk.cs25backend.security;

import edu.dongguk.cs25backend.domain.type.UserRole;
import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Collections;
import java.util.List;


@Getter
public class CustomUserDetail implements UserDetails {
    private Long id;
    private Collection<? extends GrantedAuthority> authorities;


    public CustomUserDetail(Long id,Collection<? extends GrantedAuthority> authorities) {
        this.id = id;
        this.authorities = authorities;
    }

    public static CustomUserDetail create(UserLoginForm user) {
        List<GrantedAuthority> authorities = Collections.
                singletonList(new SimpleGrantedAuthority(user.getRole().toString()));

        return new CustomUserDetail(
                user.getId(),
                authorities
        );
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return authorities;
    }

    @Override
    public String getPassword() {
        return null;
    }

    @Override
    public String getUsername() {
        return id.toString();
    }

    @Override
    public boolean isAccountNonExpired() {
        return false;
    }

    @Override
    public boolean isAccountNonLocked() {
        return false;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return false;
    }

    @Override
    public boolean isEnabled() {
        return false;
    }
}

File: cs25server/src/main/kotlin/edu/dongguk/cs25server/security/CustomUserDetail.kt
package edu.dongguk.cs25server.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class CustomUserDetail (
    private val id: Long,
    private var authorities: List<GrantedAuthority>
    ) : UserDetails {

    override fun getAuthorities() = authorities

    override fun getPassword() = null

    override fun getUsername() = id.toString()

    override fun isAccountNonExpired() = false

    override fun isAccountNonLocked() = false

    override fun isCredentialsNonExpired() = false

    override fun isEnabled() = false

    companion object {
        fun create(user: UserLoginForm): CustomUserDetail {
            val authorities = listOf<GrantedAuthority>(SimpleGrantedAuthority("ROLE_" + user.getRole()))
            return CustomUserDetail(id = user.getId(), authorities = authorities)
        }
    }
}